presto-main/src/main/java/io/prestosql/metadata/FunctionRegistry.java

```diff
@@ -910,7 +911,30 @@ private boolean returnsNullOnGivenInputTypes(ApplicableFunction applicableFuncti
 
     public FunctionMetadata getFunctionMetadata(ResolvedFunction resolvedFunction)
     {
-        return getSpecializedFunctionKey(resolvedFunction).getFunction().getFunctionMetadata();
+        FunctionMetadata functionMetadata = getSpecializedFunctionKey(resolvedFunction).getFunction().getFunctionMetadata();
+
+        // specialize function metadata to resolvedFunction
+        List<FunctionArgumentDefinition> argumentDefinitions;
+        if (functionMetadata.getSignature().isVariableArity()) {
+            List<FunctionArgumentDefinition> fixedArguments = functionMetadata.getArgumentDefinitions().subList(0, functionMetadata.getArgumentDefinitions().size() - 1);
+            int variableArgumentCount = resolvedFunction.getSignature().getArgumentTypes().size() - fixedArguments.size();
```

Comment:
I assume this is because the signature of the resolve function has all variable arguments reified? Otherwise it would always be `1`?

Reply:
Yes, but it would be zero in that case.